relayer/chains/evm/provider.go

```diff
@@ -249,13 +252,22 @@ func (p *Provider) GetTransationOpts(ctx context.Context) (*bind.TransactOpts, e
 	if err != nil {
 		return nil, err
 	}
+	// lock here to prevent transcation replacement
+	globalOptsLock.Lock()
+	defer globalOptsLock.Unlock()
+
 	txOpts.Nonce = p.NonceTracker.Get(wallet.Address)
```

Comment:
Why are we not using nounceAt here aswell?

Reply:
We cannot reliably get correct nonce when trans are executing. I tried with that too. but still had mishaps.